From 5c995f8ebf22a790374bf1964c1cf135c132842d Mon Sep 17 00:00:00 2001
From: nkhan <nkhan@nvidia.com>
Date: Tue, 11 Feb 2025 15:47:02 -0800
Subject: [PATCH] * genromfs.c (alignnode): close(fd) after read

Issue:
In MSYS2, when processing a large set of files (approximately 9,000), genromfs
displays the error:

"Too many open files"

Fix:
To resolve this, the open file handles are now explicitly closed after reading,
preventing the system from exceeding the file descriptor limit.

Signed-off-by: Noman Khan <nkhan@nvidia.com>
---
 genromfs.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/genromfs.c b/genromfs.c
index a804d37..e81dbfc 100644
--- a/genromfs.c
+++ b/genromfs.c
@@ -632,6 +632,7 @@ int alignnode(struct filenode *node, int curroffset, int extraspace)
 			if ((fd = open(node->realname, O_RDBIN)) >= 0) {
 				memset(bigbuf, 0, 16);
 				read(fd, bigbuf, 16);
+				close(fd);
 				checkpos = 0;
 			}
 		} else if (S_ISLNK(node->modes)) {
-- 
2.47.1

